dpdispatcher/contexts/local_context.py

```diff
@@ -208,7 +208,10 @@ def download(
                             shutil.rmtree(lfile, ignore_errors=True)
                         elif os.path.isfile(lfile) or os.path.islink(lfile):
                             os.remove(lfile)
-                        shutil.copyfile(rfile, lfile)
+                        if not os.path.islink(rfile):
+                            shutil.move(rfile, lfile)
+                        else:
+                            shutil.copyfile(rfile, lfile)
                     else:
                         raise RuntimeError("should not reach here!")
```

Comment:
The comment "should not reach here!" is not very informative. Consider providing more context or removing the comment if it's not necessary. [medium]